src/k2/transport/RetryStrategy.h

```diff
@@ -97,13 +97,11 @@ class ExponentialBackoffStrategy {
                         _success = !fut.failed();
                         resultPtr->ignore_ready_future(); // ignore previous result stored in the result
                         (*resultPtr.get()) = std::move(fut);
-                        K2LOG_D(log::txretry, "round {} ended with success={}", _try, _success);
-                        if (!_success) {
-                            _try++;
-                            _currentTimeout = Duration((uint64_t)(nsec(_currentTimeout).count() * _rate));
-                            if (!deadline.isOver()) {
-                                return seastar::sleep(deadline.getRemaining());
-                            }
+                        K2LOG_D(log::tx, "round ended with success={}", _success);
+                        if (!_success && !deadline.isOver()) {
```

Comment:
Looks like this is reverting Ivan's previous fix. Is that intentional or left over from the merge problems?

Reply:
It looks like a merging problem. I'll fix manually.